codegen/lib/src/field_validator_factory.dart

```diff
@@ -304,3 +317,7 @@ typedef Expression LazyViolation(Expression fieldValue);
 ///  - `false` if the constraint obeyed.
 ///
 typedef Expression LazyCondition(Expression fieldValue);
+
+/// A function of a field value expresion to code which prepares context before the constraint
```

Comment:
`expression`?

And honestly, I don't understand.
Can you please explain in more detail and maybe have an example?